src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -0,0 +1,225 @@
+            var razorFileUri = request.Uris.Last();
+            var fileName = Path.GetFileName(razorFileUri.GetAbsoluteOrUNCPath());
+            if (!Path.GetExtension(fileName).Equals(".razor", StringComparison.OrdinalIgnoreCase))
+            {
+                _logger.LogInformation("Last file in the drop was not a single razor file URI.");
+                return null;
+            }
+
+            if (request.Uris.Any(uri => !Path.GetFileName(uri.GetAbsoluteOrUNCPath()).StartsWith(fileName)))
+            {
+                _logger.LogInformation("One or more URIs were not a child file of the main .razor file.");
+                return null;
+            }
+
+            var componentTagText = await TryGetComponentTagAsync(razorFileUri, cancellationToken).ConfigureAwait(false);
+            if (componentTagText is null)
+            {
+                return null;
+            }
+
+            return new WorkspaceEdit
+            {
+                DocumentChanges = new Container<WorkspaceEditDocumentChange>(
+                    new WorkspaceEditDocumentChange(
+                        new TextDocumentEdit
+                        {
+                            TextDocument = new()
+                            {
+                                Uri = request.TextDocument.Uri
+                            },
+                            Edits = new[]
+                            {
+                                    new TextEdit
+                                    {
+                                        NewText = componentTagText,
+                                        Range = request.Range
+                                    }
+                            }
+                        }
+                    )
+                )
+            };
+        }
+
+        private async Task<string?> TryGetComponentTagAsync(Uri uri, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Trying to find document info for dropped uri {uri}.");
+
+            var documentSnapshot = await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(() =>
+            {
+                _documentResolver.TryResolveDocument(uri.GetAbsoluteOrUNCPath(), out var documentSnapshot);
+
+                return documentSnapshot;
+            }, cancellationToken).ConfigureAwait(false);
+
+            if (documentSnapshot is null)
+            {
+                _logger.LogWarning($"Failed to find document {uri}.");
+                return null;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var descriptor = await _razorComponentSearchEngine.TryGetTagHelperDescriptorAsync(documentSnapshot).ConfigureAwait(false);
+            if (descriptor is null)
+            {
+                _logger.LogWarning($"Failed to find tag helper descriptor.");
+                return null;
+            }
+
+            var typeName = descriptor.GetTypeNameIdentifier();
```

Comment:
@javiercn would you mind just quickly looking over how we're translating a TagHelperDescriptor => HTML tag real quick? Hoping to validate that we're using the correct properties of the TagHelper descriptor for the attributes and element names etc.